cnf-certification-test/performance/suite.go

```diff
@@ -72,13 +72,20 @@ func testSchedulingPolicyInCPUPool(env *provider.TestEnvironment,
 		pids := crclient.GetPidsFromPidNamespace(pidNamespace, testContainer)
 
 		// Check for the specified priority for each processes running in that pid namespace
-		if scheduling.ProcessPidsCPUScheduling(pids, testContainer, nonCompliantContainers, schedulingType) {
+		if scheduling.ProcessPidsCPUScheduling(pids, testContainer, nonCompliantContainersMap, schedulingType) {
 			compliantContainers = append(compliantContainers, testContainer)
 		}
 		logrus.Debugf("Processed %v", testContainer)
 	}
-	if len(nonCompliantContainers) != 0 {
+	if len(nonCompliantContainersMap) != 0 {
+		nonCompliantContainers := []*provider.Container{}
+		for k := range nonCompliantContainersMap {
+			nonCompliantContainers = append(nonCompliantContainers, k)
+		}
 		testhelper.AddTestResultLog("Non-compliant", nonCompliantContainers, tnf.ClaimFilePrintf, ginkgo.Fail)
 	}
+	if len(compliantContainers) == 0 {
```

Comment:
I also agree, either we have no container to test, then we skip the test (upstream of this point) or we have at least 1 container failing, then we fail the test. if all container succeed (since number of containers have to be >0 by now) then we pass the test.

Reply:
I am checking at first if there exists any non-compliant containers. And those are reported by calling testhelper.AddTestResultLog with ginkgo.Fail. After that this check is continued. It seems to me that this check is redundant.